plugins/modules/redfish_firmware_rollback.py

```diff
@@ -0,0 +1,360 @@
+                                                    api_timeout=120)
+    previous_component = list(filter(lambda d: d["Id"].startswith("Previous"), inventory_uri_resp.json_data["Members"]))
+    if not previous_component:
+        module.fail_json(msg=NO_COMPONENTS)
+    component_name = module.params["name"]
+    try:
+        component_compile = re.compile(r"^{0}$".format(component_name))
+    except sre_constants.error:
+        component_compile = ""
+        module.exit_json(msg=NO_CHANGES_FOUND)
+    prev_uri, reboot_uri = {}, []
+    for each in previous_component:
+        available_comp = each["Name"]
+        available_name = re.match(component_compile, available_comp)
+        if not available_name:
+            continue
+        if available_name.group() in REBOOT_COMP:
+            reboot_uri.append(each["@odata.id"])
+            continue
+        prev_uri[each["Version"]] = each["@odata.id"]
+    if module.check_mode and (prev_uri or reboot_uri):
+        module.exit_json(msg=CHANGES_FOUND, changed=True)
+    elif (module.check_mode or not module.check_mode) and (not prev_uri and not reboot_uri):
+        module.exit_json(msg=NO_CHANGES_FOUND)
+    return list(prev_uri.values()), reboot_uri, update_uri
+
+
+def get_job_status(redfish_obj, module, job_ids, job_wait=True):
+    each_status, failed_count, js_job_msg = [], 0, ""
+    wait_timeout = module.params["reboot_timeout"]
+    for each in job_ids:
+        each_job_uri = MANAGER_JOB_ID_URI.format(each)
+        job_resp, js_job_msg = wait_for_redfish_job_complete(redfish_obj, each_job_uri, job_wait=job_wait,
+                                                             wait_timeout=wait_timeout)
+        if job_resp and js_job_msg:
+            module.exit_json(msg=JOB_WAIT_MSG.format(wait_timeout), job_status=[strip_substr_dict(job_resp.json_data)],
+                             changed=True)
+        job_status = job_resp.json_data
+        if job_status["JobState"] == "Failed":
+            failed_count += 1
+        strip_odata = strip_substr_dict(job_status)
+        each_status.append(strip_odata)
+    return each_status, failed_count
+
+
+def require_session(idrac, module):
+    session_id, token = "", None
+    payload = {'UserName': module.params["username"], 'Password': module.params["password"]}
+    path = SESSION_RESOURCE_COLLECTION["SESSION"]
+    resp = idrac.invoke_request('POST', path, data=payload, api_timeout=120)
+    if resp and resp.success:
+        session_id = resp.json_data.get("Id")
+        token = resp.headers.get('X-Auth-Token')
+    return session_id, token
+
+
+def wait_for_redfish_idrac_reset(module, redfish_obj, wait_time_sec, interval=30):
+    time.sleep(interval // 2)
+    msg = RESET_UNTRACK
+    wait = wait_time_sec
+    track_failed = True
+    resetting = False
+    while wait > 0 and track_failed:
+        try:
+            redfish_obj.invoke_request("GET", MANAGERS_URI, api_timeout=120)
```

Comment:
do we need to make api_timeout hardcoded ?

Reply:
We have `timeout` default to `30` some of the iDRAC8 require more `api_timeout` value so increased it to `120`